tcms/static/js/tcms_actions.js

```diff
@@ -1392,20 +1391,25 @@ function exportCase(url, form, table) {
 
 var printableCases = exportCase;
 
-function validateIssueID(bugSystemId, bugId) {
-  if (bugSystemId == 1) {
-    var result = /^\d{1,7}$/.test(bugId);
-    if (!result) {
-      window.alert(default_messages.alert.invalid_bug_id);
-    }
-    return result;
-  } else if (bugSystemId == 2) {
-    var result = /^[A-Z0-9]+-\d+$/.test(bugId);
-    if (!result) {
-      window.alert(default_messages.alert.invalid_jira_id);
-    }
-    return result;
+function validateIssueID(bugRegExp, bugId) {
+  // if bugRegExp is empty string then all input is valid!
+  if (!bugRegExp) {
+    return true;
+  }
+
+  try {
+    var bug_re = new RegExp(bugRegExp);
+  // catch syntax errors in regular expressions
+  } catch(err) {
+    window.alert(err.name + ': ' + err.message);
```

Comment:
In whatever the case, it is not a good idea to display such underlying error message to user in client side. Instead, a descriptive message should be displayed.